app/src/main/java/com/pvp/app/api/ExerciseService.kt

```diff
@@ -26,4 +26,10 @@ interface ExerciseService {
      * @return a list of sport activities that the user does not participate in frequently
      */
     suspend fun getInfrequentActivities(maxOccurrence: Int = 1): List<SportActivity>
+
+    /**
+     * Reads user activity data from HealthConnect and finds the most frequently done activity
+     * @return the most frequently done activity
+     */
+    suspend fun getMostFrequentActivity(): SportActivity
```

Comment:
If there is none, i.e. user has a fresh phone/google account? What happens? Maybe we should fallback to `Walking` activity?

Reply:
Yes, that's exactly what happens. Will expand the comment to include that.